api/app/Models/AssessmentStep.php

```diff
@@ -90,18 +90,27 @@ protected static function boot()
         parent::boot();
 
         static::creating(function (AssessmentStep $step) {
+
+            $poolsExistingSteps = AssessmentStep::where('pool_id', $step['pool_id'])->orderBy('sort_order', 'desc')->get();
+
             if (isset($step['type']) && $step['type'] === AssessmentStepType::APPLICATION_SCREENING->name) {
                 $sortOrder = 1;
             } elseif (isset($step['type']) && $step['type'] === AssessmentStepType::SCREENING_QUESTIONS_AT_APPLICATION->name) {
+                if (! is_null($poolsExistingSteps->where('sort_order', 2)->first())) {
+                    // a step is currently sitting in the second slot, push 'em all down first
+                    foreach ($poolsExistingSteps as $existingStep) {
+                        if ($existingStep->sort_order !== 1) {
+                            $current = $existingStep->sort_order;
+                            $existingStep->sort_order = $current + 1;
+                            $existingStep->save();
+                        }
+                    }
+                }
                 $sortOrder = 2;
             } else {
-                $highestSortOrderStep = AssessmentStep::where('pool_id', $step['pool_id'])->orderBy('sort_order', 'desc')->first();
-                $highestSortOrder = isset($highestSortOrderStep) ? $highestSortOrderStep->sort_order : 0;
-                if ($highestSortOrder < 3) {
-                    $sortOrder = 3;
```

Comment:
Probably the main thing, the second step being defaulted to sort order three. Likely the source for issues, as you'd have the application screening as step 1 then the next step as sort order 3 but nothing in between